homeassistant/components/tuya/cover.py

```diff
@@ -229,21 +230,32 @@ def __init__(
                 device.status_range[tilt_dpcode].values
             )
 
+        # Determine current_position DPCodes
+        if isinstance(self.entity_description.current_position, DPCode):
+            self._position_dpcode = (
+                self.entity_description.current_position
+                or self.entity_description.set_position
+            )
```

Comment:
This can never be `set_postition`?

Reply:
I was thinking to change them like this code?
```
        # Determine current_position DPCodes
        if isinstance(self.entity_description.current_position, DPCode):
            if self.entity_description.current_position is not None:
                self._position_dpcode = self.entity_description.current_position
            elif self.entity_description.set_position is not None:
                self._position_dpcode = self.entity_description.set_position
            else:
                self._position_dpcode = None
```

Reply:
or i should use this
```
        # Determine current_position DPCodes
        if self.entity_description.current_position is None:
            if self.entity_description.set_position is not None:
                self._position_dpcode = self.entity_description.set_position
            else:
                self._position_dpcode = None
        elif isinstance(self.entity_description.current_position, DPCode):
                self._position_dpcode = self.entity_description.current_position
        elif isinstance(self.entity_description.current_position, tuple):
            self._position_dpcode = next(
                (
                    dpcode
                    for dpcode in self.entity_description.current_position
                    if self.device.status.get(dpcode) is not None
                ),
                None,
            )
```